### PR TITLE
Hive Status and Tunnel List Info Now Includes Distance

### PR DESCRIPTION
**Pull request by Surrealaser**
## About The Pull Request

1. Distance data is now included for Hive and Tunnel status lists.

2. Should fix a bug that stopped Tunnel lists from being openable.

## Why It's Good For The Game

1. Distance data is now included for Hive and Tunnel status lists.

2. Should fix a bug that stopped Tunnel lists from being openable.

## Changelog
:cl:
code: Distance data is now included for Hive and Tunnel status lists.
fix: Should fix a bug that stopped Tunnel lists from being openable.
/:cl:

### CODE REVIEW

**Comment by BraveMole** [comment]
Since you're here, tunnel created when making a silo are missing a " "(space character) between tunnel and resin silo
can you fix? Didn't want do make a PR just for that

File: code/modules/mob/living/carbon/xenomorph/xenoprocs.dm [hunk 1]
```diff
@@ -16,24 +16,14 @@
...
-			hive = xeno_user.hive
-	else
-		hive = GLOB.hive_datums[XENO_HIVE_NORMAL]
-
-	if(!hive)
-		CRASH("couldnt find a hive in check_tunnel_list")
-
 	dat += "<b>List of Hive Tunnels:</b><BR>"
 
 	for(var/obj/structure/tunnel/T in GLOB.xeno_tunnels)
```
Comment (SplinterGP) [inline code/modules/mob/living/carbon/xenomorph/xenoprocs.dm:21]:
```suggestion
	for(var/obj/structure/tunnel/T AS in GLOB.xeno_tunnels)
```

**Review by SplinterGP — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/xeno.dm [hunk 2]
```diff
@@ -644,6 +644,7 @@ TUNNEL
 	max_integrity = 140
 	var/mob/living/carbon/xenomorph/hivelord/creator = null
 
+	var/hivenumber = XENO_HIVE_NORMAL
 	hud_possible = list(XENO_TACTICAL_HUD)
```
Comment (TiviPlus) [inline code/game/objects/structures/xeno.dm:648]: Declare new vars after redifining old ones

File: code/modules/mob/living/carbon/xenomorph/xenoprocs.dm [hunk 3]
```diff
@@ -57,9 +47,11 @@
 			if(0 to 0.33)
 				hp_color = "red"
 
+		var/distance = get_dist(user, X)
+
 		xenoinfo += " <b><font color=[hp_color]>Health: ([X.health]/[X.maxHealth])</font></b>"
 
-		xenoinfo += " <b><font color=green>([AREACOORD_NO_Z(X)])</b></td></tr>"
+		xenoinfo += " <b><font color=green>([AREACOORD_NO_Z(X)][distance > 0 ? " <b>Distance: [distance]</b>" : ""])</font></b></td></tr>"
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/xenoprocs.dm:54]: Why is this here move it to right next to where it's used or even better just inline it

File: code/modules/mob/living/carbon/xenomorph/xenoprocs.dm [hunk 4]
```diff
@@ -16,24 +16,14 @@
...
-
 	dat += "<b>List of Hive Tunnels:</b><BR>"
 
-	for(var/obj/structure/tunnel/T in GLOB.xeno_tunnels)
-		if(T.creator.hive == hive)
-			dat += "<b>[T.name]</b> located at: <b><font color=green>([T.tunnel_desc])</b></font><BR>"
+	for(var/obj/structure/tunnel/T AS in GLOB.xeno_tunnels)
+		if(user.issamexenohive(T))
+			var/distance = get_dist(user, T)
+			dat += "<b>[T.name]</b> located at: <b><font color=green>([T.tunnel_desc][distance > 0 ? " <b>Distance: [distance])</b>" : ""]</b></font><BR>"
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/xenoprocs.dm:24]: Why not inline this

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/xenomorph/xenoprocs.dm (hunk 3 above)
Comment (Surrealaser) [reply code/modules/mob/living/carbon/xenomorph/xenoprocs.dm:54]: I can if you insist, but it's used twice, so it would have to be inlined twice.

File: code/modules/mob/living/carbon/xenomorph/xenoprocs.dm (hunk 4 above)
Comment (Surrealaser) [reply code/modules/mob/living/carbon/xenomorph/xenoprocs.dm:24]: As above.

File: code/game/objects/structures/xeno.dm (hunk 2 above)
Comment (TiviPlus) [reply code/game/objects/structures/xeno.dm]: This is stilll standing

**Review by TiviPlus — APPROVED** [review]
